src/next/main.ts

```diff
@@ -0,0 +1,143 @@
+import * as fse from 'fs-extra';
+import * as uuid from "uuid";
+import TelemetryReporter from "vscode-extension-telemetry";
+import { ErrorCode } from './ErrorCode';
+import { EventName } from "./EventName";
+
+let _isDebug: boolean = false;
+let reporter: TelemetryReporter;
+
+/**
+ * Initialize TelemetryReporter by parsing attributes from a JSON file.
+ * It reads these attributes: publisher, name, version, aiKey.
+ * @param jsonFilepath absolute path of a JSON file.
+ */
+export async function initilizeFromJsonFile(jsonFilepath: string, _debug?: boolean): Promise<void> {
+    if (await fse.pathExists(jsonFilepath)) {

```

Comment:
I myself prefer to provide "else" block for every "if" block for clear boundary of the branch code, it can let me make less mistakes. But  I'm not stick to that and I'm ok with either of them. And your style looks more concise.